examples/experimental/terrain/src/surface.js

```diff
@@ -5,17 +5,29 @@ const SATELLITE = 'https://api.mapbox.com/v4/mapbox.satellite';
 // Set your mapbox token here
 const MAPBOX_TOKEN = process.env.MapboxAccessToken; // eslint-disable-line
 
-export const getSurface = ({x, y, z}, surface) => {
+export const getSurface = (surface, tile) => {
   let surfaceImage = null;
   switch (surface) {
     case 'sectional':
-      surfaceImage = `${SECTIONAL}/${z}/${x}/${y}.png?origin=nw`;
+      surfaceImage = `${SECTIONAL}/{z}/{x}/{y}.png?origin=nw`;
+      if (tile) {
+        const {x, y, z} = tile;
+        surfaceImage = `${SECTIONAL}/${z}/${x}/${y}.png?origin=nw`;
+      }
       break;
     case 'satellite':
-      surfaceImage = `${SATELLITE}/${z}/${x}/${y}@2x.png?access_token=${MAPBOX_TOKEN}`;
+      surfaceImage = `${SATELLITE}/{z}/{x}/{y}@2x.png?access_token=${MAPBOX_TOKEN}`;
+      if (tile) {
+        const {x, y, z} = tile;
+        surfaceImage = `${SATELLITE}/${z}/${x}/${y}@2x.png?access_token=${MAPBOX_TOKEN}`;
+      }
       break;
     case 'street':
-      surfaceImage = `${STREET}/${z}/${x}/${y}.png`;
+      surfaceImage = `${STREET}/{z}/{x}/{y}.png`;
+      if (tile) {
+        const {x, y, z} = tile;
+        surfaceImage = `${STREET}/${z}/${x}/${y}.png`;
+      }
       break;
     case 'none':
     default:
```

Comment:
The reason `<string> -> null` does not work has to do with the `texture` prop is async and default to `null`. The layer does not think the prop has changed. The fix will not be trivial. As an work around, you can set `surfaceImage` to `0`.

Reply:
Ok - good enough for me :) thanks for looking. I’ll just add a comment to document this.

Reply:
Hmm, this didn't work. I'm pushing up what I tried.